core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -53,6 +53,10 @@ private function addMissingTablesNeededForJoins()
             if (!$logTable->getColumnToJoinOnIdVisit()) {
                 $tableNameToJoin = $logTable->getLinkTableToBeAbleToJoinOnVisit();
 
+                if (empty($tableNameToJoin) && $logTable->getWaysToJoinToOtherLogTables()) {
```

Comment:
I'm not 100% sure about this... wouldn't we need to bubble up all ways to join other log tables to see if it can  be joined with the original table eventually? Eg 

`custom_table.userid` => refers to `custom_table2.userid` but this one has no `idvisit` ... then we shouldn't add those tables etc and not assume they can be joined when otherwise only `log_link_visit_action` is present. It may be edge case and should pretty much only happen though when not configuring it properly maybe or when creating log tables that don't make sense in terms of Matomo?

Reply:
or is it not too important because the query will then fail anyway?

Reply:
I guess the query would fail if something is configured badly. E.g. if a plugin defines dimensions and/or segments for a log table that is not joinable with log_visit or log_action. But I'll have a thought if there's a smart way to check if bubbling up ends in log_visit or log_action

Reply:
@tsteur I tried starting to implement a way to automatically add all tables up the hierarchy
It gets quite more complicated though. See
https://github.com/matomo-org/matomo/compare/improvejoins...improvejoins2?expand=1

Not sure if it's worth the effort. Otherwise it would only work if a log table defines another way to directly join into `log_visit` or `log_action`

Reply:
OK maybe not too important. I reckon could have been also useful for custom reports to know whether a table can be joined or not in order to eg know which dimensions can be combined etc. I suppose it might not be too important for now.

Reply:
So how to proceed here now? Adding the table or not?

Reply:
@tsteur Did some more testing on this. Seems its needed to add required tables up the hierarchy. Otherwise archiving for all reports including conversions fails, as it's not possible to join the custom log table with log_conversion when using a segment for the custom log table. Will merge the other branch here. Would be good if you could do another review then. Will also try to add some more specific tests